### ISSUE TITLE
ContentBundle/Teaser does not using locale when searching documents

### ISSUE THREAD

**Issue by sergiy-scherbinin** [sulu/sulu#3434]
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.5+, 1.6+
| Browser Version | -

#### Actual Behavior

Teaser block skip current chosen locale
ContentTeaserProvider - does not take into account locale when searching documents

#### Expected Behavior

Teaser block showing documents in current chosen locale
ContentTeaserProvider - take into account locale when searching documents (takes current locale)

#### Steps to Reproduce

1. create document in several locales (example: EN & DE)
2. create document with teaser block in several locales EN & DE
3. choose created (on first step) document in teaser block

Current result:
* teaser block shows document in one locale (for an example EN) for both documents (EN & DE) with teaser block

Shold be:
* teaser block should contain document in same locale

#### Possible Solutions

Include locale when searching documents for teaser block

https://github.com/sulu/sulu/blob/1.5.4/src/Sulu/Bundle/ContentBundle/Teaser/ContentTeaserProvider.php#L60-L63

```diff
@@ -60,6 +60,7 @@
         $searchResult = $this->searchManager
             ->createSearch(implode(' OR ', $statements))
             ->indexes($this->getPageIndexes())
+            ->locale($locale)
             ->execute();
 
         /** @var QueryHit $item */
```

**Comment by wachterjohannes** [comment]
@sergiy-scherbinin thanks for your issue :) I have to investigate this issue in more detail - because the fix you posted looks good but I think that we will have some troubles with `ghost` and `shadow` pages. maybe you could create a PR with this fix - then we would see if the tests are green

**Comment by trickreich** [comment]
@wachterjohannes Let's talk about this at monday. I think we have to fix this.

**Comment by danrot** [comment]
fixed in #3739
